packages/airnode-deployer/src/infrastructure/index.ts

```diff
@@ -161,8 +161,11 @@ async function terraformAirnodeManage(
     'no-color',
   ];
 
-  if (httpGatewayApiKey) {
-    commonArguments.push(['var', 'api_key', httpGatewayApiKey]);
+  if (httpGateway?.enabled) {
```

Comment:
This function handles both Airnode deploy and removal - It would be nicer if there were two functions for this and they would use a function for common parts (common arguments).

You could also better TS type this.

Reply:
Yes, I was thinking about that too. I'll make a refactor issue and do that during the 0.6 window.

Reply:
https://api3dao.atlassian.net/browse/AN-545